src/Uno.UI.Composition/Composition/Visual.skia.cs

```diff
@@ -190,10 +178,17 @@ private void Render(in PaintingSession parentSession)
 
 		using (var session = CreateLocalSession(in parentSession))
 		{
+			var canvas = session.Canvas;
+
+			ApplyPrePaintingClipping(canvas);
+
+			// Rendering shouldn't depend on matrix or clip adjustments having in a visual's Paint. That should
+			// be specific to that visual and should not affect the rendering of any other visual.
+			var count = canvas.Save();
```

Comment:
Why do we need this? `Paint` should not alter the `Canvas` (and if it does, it should be its responsibility to restore it IMO).

Reply:
We don't. It's more of a guard in case someone alters the Canvas and forgot to set it back. Actually, now thinking about it, I would prefer to assert against the SaveCount.